bintool/metadata.h

```diff
@@ -255,7 +255,9 @@ struct partition_table_item : public single_byte_size_item {
             new_p.flags = permissions_flags & (~PICOBIN_PARTITION_PERMISSIONS_BITS);
 
             if (new_p.flags & PICOBIN_PARTITION_FLAGS_HAS_ID_BITS) {
-                new_p.id = (uint64_t)data[i++] | ((uint64_t)data[i++] << 32);
+                uint32_t low = data[i++];
+                uint32_t high = data[i++];
+                new_p.id = (uint64_t)low | ((uint64_t)high << 32);
```

Comment:
Alternatively, I guess you could do
```c
new_p.id = (uint64_t)data[i] | ((uint64_t)data[i+1] << 32);
i += 2;
```
which would avoid the temporary variables?

Reply:
I think I prefer the original style - using multiple `i++` matches the style used throughout the rest of the file